morebits.js

```diff
@@ -302,7 +301,7 @@ Morebits.quickForm.element.prototype.compute = function QuickFormElementCompute(
 	var node;
 	var childContainder = null;
 	var label;
-	var id = (in_id ? in_id + '_' : '') + 'node_' + this.id;
+	var id = (in_id ? in_id + '_' : '') + 'node_' + Morebits.quickForm.element.id++;
```

Comment:
So, this means that simply doing `new Morebits.quickForm.element({data})` no longer results in an `id`, since that's not created until here.  I'm not sure that matters? Maybe in theory for `getElements`?

I wonder if it wouldn't be better to make use of the second parameter to `compute` in both these cases?  It would mean not incrementing the value numerically, but is this not what `in_id` is for?  Alternatively, what about explicitly supplying the new `data` with `Morebits.quickForm.element.id++`?  I've not tested either yet.

Reply:
This approach feels better in theory - every time we enter compute we are going to have an HTMLElement by the end of it with a unique id. The 2nd param of compute() is only used in radio/checkbox subgroups. Using that seems like a more complicated approach, as we don't care about what the IDs are - just that they be distinct. 

>So, this means that simply doing new Morebits.quickForm.element({data}) no longer results in an id, since that's not created until here. I'm not sure that matters? Maybe in theory for getElements?
 
it shouldn't matter at all.

Reply:
Agree that it really doesn't matter since there's no real meaning to any of it, but semantically, I would think it'd make sense for the optgroup and options to appear with some connection to the parent.  At any rate, a quick test with `in_id` didn't work, so I'm not sure it's actually doing what I think it should.  Maybe down the road but clearly no need to deal with it now.  I did have luck with just setting the id explicitly, but I suppose that's no real difference!